### ISSUE TITLE
Device configuration not sharable

### ISSUE THREAD

**Issue by MattHag** [pwr-Solaar/Solaar#2453]
**Information**
- Solaar version: 1.1.12

**Is your feature request related to a problem? Please describe.**
I'm always frustrated when I switch between the same mouse model, and my rule do not work anymore. The mouse and keyboard specific settings can not be stored and retrieved, nor shared.

**Describe the solution you'd like**
Store device specific configuration, so that it can be stored and shared with others. Sharing a file should allow anyone with e.g. an MX Master 3 to load the configuration and use the combination of a device and rule configuration for that device. No need to deal with button mappings or diversion for each unique device.

**Describe alternatives you've considered**
Manually skimming through the GUI and settings up buttons, to make the rules work.

**Additional context**
It should be possible to share a config for a mouse (e.g. MX Master 3), so it can be used for any other MX Master 3 out there.

**Comment by MattHag** [comment]
Yes, this needs a good foundation and assumptions for the future of rules. Rules should probably be device dependent by default in order to simplify further development. That's common for peripheral manufacturers' official software.

**Comment by pfps** [comment]
PR #2951 adds Device and Active rule conditions matching on device names, so rules can be specific to a particular model (more or less).

The rest of this is unlikely to be implemented.
